src/energyplus/ForwardTranslator/ForwardTranslateZoneHVACEvaporativeCoolerUnit.cpp

```diff
@@ -0,0 +1,247 @@
+#include <utilities/idd/Fan_SystemModel_FieldEnums.hxx>
+#include <utilities/idd/EvaporativeCooler_Direct_ResearchSpecial_FieldEnums.hxx>
+#include <utilities/idd/EvaporativeCooler_Indirect_ResearchSpecial_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+
+using namespace openstudio::model;
+
+namespace openstudio {
+
+namespace energyplus {
+
+  boost::optional<IdfObject> ForwardTranslator::translateZoneHVACEvaporativeCoolerUnit(model::ZoneHVACEvaporativeCoolerUnit& modelObject) {
+
+    // Instantiate an IdfObject of the class to store the values
+    IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::ZoneHVAC_EvaporativeCoolerUnit, modelObject);
+
+    // Availability Schedule Name: Optional Object
+    if (boost::optional<Schedule> availabilitySchedule_ = modelObject.availabilitySchedule()) {
+      if (boost::optional<IdfObject> wo_ = translateAndMapModelObject(availabilitySchedule_.get())) {
+        idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::AvailabilityScheduleName, wo_->nameString());
+      }
+    }
+
+    boost::optional<std::string> outdoorAirInletNodeName;
+    boost::optional<std::string> coolerOutletNodeName;
+
+    // Outdoor Air Inlet Node Name: Required Node
+    if (boost::optional<Node> node = modelObject.inletNode()) {
+      outdoorAirInletNodeName = node->name().get();
+      idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::OutdoorAirInletNodeName, node->name().get());
+    }
+
+    // Cooler Outlet Node Name: Required Node
+    if (boost::optional<Node> node = modelObject.outletNode()) {
+      coolerOutletNodeName = node->name().get();
+      idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::CoolerOutletNodeName, node->name().get());
+    }
+
+    // Supply Air Fan Object Type
+    // Supply Air Fan Name
+    boost::optional<IdfObject> fan_;
+    if (boost::optional<HVACComponent> supplyAirFan = modelObject.supplyAirFan()) {
+      fan_ = translateAndMapModelObject(supplyAirFan.get());
+
+      if (fan_ && fan_->name()) {
+        idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::SupplyAirFanObjectType, fan_->iddObject().name());
+        idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::SupplyAirFanName, fan_->name().get());
+      }
+    }
+
+    if (modelObject.isDesignSupplyAirFlowRateAutosized()) {
+      idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::DesignSupplyAirFlowRate, "Autosize");
+    } else {
+      // Design Supply Air Flow Rate: boost::optional<double>
+      if (boost::optional<double> designSupplyAirFlowRate_ = modelObject.designSupplyAirFlowRate()) {
+        idfObject.setDouble(ZoneHVAC_EvaporativeCoolerUnitFields::DesignSupplyAirFlowRate, designSupplyAirFlowRate_.get());
+      }
+    }
+
+    // Fan Placement: Required String
+    const std::string fanPlacement = modelObject.fanPlacement();
+    idfObject.setString(ZoneHVAC_EvaporativeCoolerUnitFields::FanPlacement, fanPlacement);
```

Comment:
(nitpick: my preference is to not create an intermediate variable. That way there's no risk to reuse it later in the scope. Or at least do that inside "{}" to scope it)